### PR TITLE
{Monitor}Bug fix for stop iteration

### LINKED ISSUE #12191: Unexpected error while list autoscale rules
## Describe the bug

**Command Name**
`az monitor autoscale rule list`

**Errors:**
```
Traceback (most recent call last):
Temp\pip-install-2ombfg7z\knack\knack\cli.py, ln 206, in invoke
azure\cli\core\commands\__init__.py, ln 608, in execute
azure\cli\core\commands\__init__.py, ln 666, in _run_jobs_serially
azure\cli\core\commands\__init__.py, ln 657, in _run_job
azure\cli\command_modules\monitor\_exception_handler.py, ln 42, in monitor_exception_handler
Local\Temp\pip-install-2ombfg7z\six\six.py, ln 696, in reraise
azure\cli\core\commands\__init__.py, ln 636, in _run_job
azure\cli\core\commands\__init__.py, ln 306, in __call__
azure-cli-core\azure\cli\core\__init__.py, ln 493, in default_command_handler
cli\command_modules\monitor\operations\autoscale_settings.py, ln 341, in autoscale_rule_list
StopIteration
```

## To Reproduce:
Steps to reproduce the behavior. Note that argument values have been redacted, as they may contain sensitive information.

- _Put any pre-requisite steps here..._
- `az monitor autoscale rule list --autoscale-name {} --resource-group {}`

## Environment Summary
```
Windows-10-10.0.17763-SP0
Python 3.6.6
Shell: powershell.exe

azure-cli 2.0.78 *

```

### PR DESCRIPTION
**Pull request by MyronFanQiu**
Fix #12191 

**Description of PR (Mandatory)**  
(Why this PR? What is changed? What is the effect? etc. A high-quality description can accelerate the review process)

**Testing Guide**  
(Example commands with explanations)

**History Notes:**  
(Fill in the following template if multiple notes are needed, otherwise PR title will be used for history note.)

[Component Name 1] (BREAKING CHANGE:) (az command:) make some customer-facing change.  
[Component Name 2] (BREAKING CHANGE:) (az command:) make some customer-facing change.

---

This checklist is used to make sure that common guidelines for a pull request are followed.

- [ ] The PR title and description has followed the guideline in [Submitting Pull Requests](https://github.com/Azure/azure-cli/tree/dev/doc/authoring_command_modules#submitting-pull-requests).

- [ ] I adhere to the [Command Guidelines](https://github.com/Azure/azure-cli/blob/dev/doc/command_guidelines.md).

### CODE REVIEW

**Comment by MyronFanQiu** [comment]
@Juliehzl @arrownj @haroldrandom  Is there any better way to fix it? I mean more native in python.

File: src/azure-cli/azure/cli/command_modules/monitor/operations/autoscale_settings.py [hunk 1]
```diff
@@ -338,6 +338,10 @@ def autoscale_rule_create(cmd, client, autoscale_name, resource_group_name, cond
 
 def autoscale_rule_list(cmd, client, autoscale_name, resource_group_name, profile_name=DEFAULT_PROFILE_NAME):
     autoscale_settings = client.get(resource_group_name, autoscale_name)
+    profile_names = [x.name for x in autoscale_settings.profiles]
+    if not profile_name in profile_names:
```
Comment (haroldrandom) [inline src/azure-cli/azure/cli/command_modules/monitor/operations/autoscale_settings.py]:
```suggestion
    if profile_name not in profile_names:
```
?

File: src/azure-cli/azure/cli/command_modules/monitor/operations/autoscale_settings.py [hunk 2]
```diff
@@ -338,6 +338,10 @@ def autoscale_rule_create(cmd, client, autoscale_name, resource_group_name, cond
...
+        from knack.util import CLIError
+        raise CLIError('Profile name is invalid. Please check the existence of the profile.')
     profile = next(x for x in autoscale_settings.profiles if x.name == profile_name)
```
Comment (haroldrandom) [inline src/azure-cli/azure/cli/command_modules/monitor/operations/autoscale_settings.py]: Or, could use try...except sinppet to catch StopIteration error, then raise CLIError.

Comment (MyronFanQiu) [reply src/azure-cli/azure/cli/command_modules/monitor/operations/autoscale_settings.py]: well. So `next` cannot handle such scenario and we need extra logic. Right? Compared to try..except, I prefer current one.

File: src/azure-cli/azure/cli/command_modules/monitor/operations/autoscale_settings.py (hunk 1 above)
Comment (MyronFanQiu) [reply src/azure-cli/azure/cli/command_modules/monitor/operations/autoscale_settings.py]: fix

File: src/azure-cli/azure/cli/command_modules/monitor/tests/latest/test_monitor_autoscale.py [hunk 3]
```diff
@@ -97,6 +98,9 @@ def test_monitor_autoscale_rules(self, resource_group):
 
         # verify order is stable
         list_1 = self.cmd('monitor autoscale rule list -g {rg} --autoscale-name {vmss}').get_output_in_json()
+        with self.assertRaisesRegexp(CLIError, 'Profile name is invalid. Please check the existence of the profile.'):
+            self.cmd('monitor autoscale rule list -g {rg} --autoscale-name {vmss} --profile-name falseprofile').get_output_in_json()
```
Comment (arrownj) [inline src/azure-cli/azure/cli/command_modules/monitor/tests/latest/test_monitor_autoscale.py]: `get_output_in_json()` can be removed ?

Comment (MyronFanQiu) [reply src/azure-cli/azure/cli/command_modules/monitor/tests/latest/test_monitor_autoscale.py]: done

File: src/azure-cli/azure/cli/command_modules/monitor/operations/autoscale_settings.py (hunk 2 above)
Comment (haroldrandom) [reply src/azure-cli/azure/cli/command_modules/monitor/operations/autoscale_settings.py]: using try..except is the same what you did now.

**Pull request merged**